### PR TITLE
Use promise in tests.

### PR DESCRIPTION
**Pull request by tokuhirom**

### CODE REVIEW

File: lib/HTTP/Server/Tiny.pm6 [hunk 1]
```diff
@@ -261,6 +261,7 @@ my class HTTP::Server::Tiny::Handler {
             if $result ~~ Promise {
                 return $result.result;
             }
+            warn "P6W app must return Promise";
```
Comment (astj) [inline lib/HTTP/Server/Tiny.pm6]: P6W specs still support old psgi protocol, so returning Promise is not "`must`", in P6W specification.
https://github.com/zostay/P6W#4-application-protocol-implementation

(Btw how Crust will handle those response protocols? I think it's better not drop old psgi protocols right now)

Comment (tokuhirom) [inline lib/HTTP/Server/Tiny.pm6]: hmm? it says `An application MUST return a Promise. `.

Comment (tokuhirom) [inline lib/HTTP/Server/Tiny.pm6]: ah, we should set p6w.protocol.enabled as request-response.

Comment (tokuhirom) [inline lib/HTTP/Server/Tiny.pm6]: We'll set the value with #54.

Comment (astj) [inline lib/HTTP/Server/Tiny.pm6]: Yes. P6W app MUST return Promise when the app works in request-response protocol, but not in other protocols (like legacy psgi).

**Pull request merged**